crates/ra_hir_expand/src/builtin_macro.rs

```diff
@@ -171,3 +175,27 @@ fn file_expand(
 
     Ok(expanded)
 }
+
+fn compile_error_expand(
+    db: &dyn AstDatabase,
+    id: MacroCallId,
+    _tt: &tt::Subtree,
+) -> Result<tt::Subtree, mbe::ExpandError> {
+    let loc = db.lookup_intern_macro(id);
+    let macro_call = loc.ast_id.to_node(db);
+
+    let macro_content = {
+        // FIXME: verify that this is an ast::Literatal with kind() == LiteralKind.String, else error.
+        let arg = macro_call.token_tree().ok_or_else(|| mbe::ExpandError::UnexpectedToken)?;
```

Comment:
You could get it by using `tt` , if it is valid , it should a single subtree with one literal token-tree.

Reply:
Okay. Once I have the literal is there an existing funtion to validate it's type or do I need to manually parse the ends and look for quotes?

Reply:
As I know there is no existing function to validate it, so you may to check if it contains double quote. And if I remember correctly, the `quote!` macro support `TokenTree` too.